### PR TITLE
Fix/docker image build for production

### PR DESCRIPTION
**Pull request by BMartinos**
## Summary by CodeRabbit

- **New Features**
	- Introduced a session callback in the authentication options to enhance user state management during authentication.
	- Added necessary environment variables for database access and Keycloak integration, including configurations for NextAuth.js.

- **Bug Fixes**
	- Improved formatting and readability across various files without affecting functionality.

- **Refactor**
	- Modularized authentication configuration for better organization and maintainability.

- **Style**
	- Standardized code formatting, including the use of single quotes in import statements and improved indentation for readability.

### CODE REVIEW

**Review by yassinedorbozgithub — APPROVED** [review]
`yarn build`

image

LGTM 👍

File: .env [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
+KEYCLOAK_ADMIN=admin
+EXTERNAL_URL=http://localhost:3001
+KEYCLOAK_ADMIN_PASSWORD=password
+KEYCLOAK_CLIENT_ID="nextjs"
+KEYCLOAK_CLIENT_SECRET="gX0nfIp57bO3bY68br3kdTXung2Auwpr"
+KEYCLOAK_ISSUER="http://localhost:8082/realms/shlp"
+NEXTAUTH_URL="http://localhost:3000"
+NEXTAUTH_SECRET="ArmMS4LBmH3VTR77UrSAY2lPs04HO0Nk2/4BcU0jMvc="
+NEXT_CONTAINER_KEYCLOAK_ENDPOINT="http://localhost:8082"
+NEXT_LOCAL_KEYCLOAK_URL="http://localhost:8082"
```
[the remark this answers is not in the document]
Comment (BMartinos) [reply .env]: Fixing

File: .env [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
-POSTGRES_PASSWORD=replaceme
-KEYCLOAK_ADMIN_PASSWORD=replaceme
-KEYCLOAK_ADMIN=replaceme
-EXTERNAL_URL=replaceme
 PRISMA_FIELD_ENCRYPTION_KEY=k1.aesgcm256.DbQoar8ZLuUsOHZNyrnjlskInHDYlzF3q6y1KGM7DUN=
-PRISMA_FIELD_ENCRYPTION_HASH_SALT=replaceme
+PRISMA_FIELD_ENCRYPTION_HASH_SALT=0be97e77063ea3f7a0f128b06ef9b1eg
+POSTGRES_DB=postgres
+POSTGRES_USER=postgres
+POSTGRES_PASSWORD=postgres
```
[the remark this answers is not in the document]
Comment (BMartinos) [reply .env:5]: This is just for local development. Production credentials will be updated

File: .env [hunk 3]
```diff
@@ -7,2 +1,15 @@
...
+EXTERNAL_URL=http://localhost:3001
+KEYCLOAK_ADMIN=admin
+KEYCLOAK_ADMIN_PASSWORD=password
```
[the remark this answers is not in the document]
Comment (BMartinos) [reply .env:8]: This is just for local development. Production credentials will be updated

File: .env [hunk 4]
```diff
@@ -7,2 +1,15 @@
...
+KEYCLOAK_CLIENT_ID="nextjs"
+KEYCLOAK_CLIENT_SECRET="gX0nfIp57bO3bY68br3kdTXung2Auwpr"
```
[the remark this answers is not in the document]
Comment (BMartinos) [reply .env:10]: This is just for local development. Production credentials will be updated

**Pull request merged**